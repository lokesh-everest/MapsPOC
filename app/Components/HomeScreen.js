import React from 'react';
import {View, Button, StyleSheet, Dimensions, TouchableOpacity, Text,Image} from 'react-native'

export default class HomeScreen extends React.Component {

    handleUserClick =()=>{
        this.props.navigation.navigate('MapsContainerUser')
    }
    handleDeliveryClick=()=>{
        this.props.navigation.navigate('MapsContainerDelivery')
    }
    render() {
        return (
            <View style={{
                flex: 1,
                flexDirection: 'column',
                justifyContent: 'center',
            }}>
                <View style={{
                    flexDirection: 'row',
                    justifyContent: 'center',
                    alignItems: 'center',
                }}>
                    <TouchableOpacity style={styles.button} onPress={this.handleUserClick}>
                        <Image style={{width:50,height:50}} source={require('./../assets/home.png')} />
                            <Text style={{color: 'black'}}>User</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.button}  onPress={this.handleDeliveryClick}>
                        <Image style={{width:50,height:50}} source={require('./../assets/delievery.png')} />
                            <Text style={{color: 'black'}}>Delivery Person</Text>
                    </TouchableOpacity>
                </View>
            </View>
        )
    }
}
const styles = StyleSheet.create({
    button: {width: 150,
        height: 150,
        borderRadius: 75,
        borderWidth: 1,
        borderColor: 'red',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: "center",
        marginRight: 20
    }
})