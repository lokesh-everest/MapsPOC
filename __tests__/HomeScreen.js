import 'react-native';
import React from 'react';
import HomeScreen from "../app/Components/HomeScreen";

import renderer from 'react-test-renderer';

it('render correcly',()=>{
    const tree = renderer.create(<HomeScreen />);
    expect(tree).toMatchSnapshot();
})
