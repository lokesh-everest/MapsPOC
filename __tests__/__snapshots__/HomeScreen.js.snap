// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correcly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "borderRadius": 75,
          "borderWidth": 1,
          "flexDirection": "column",
          "height": 150,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/home.png",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        User
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "borderRadius": 75,
          "borderWidth": 1,
          "flexDirection": "column",
          "height": 150,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/delievery.png",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        Delivery Person
      </Text>
    </View>
  </View>
</View>
`;
